ERCS/erc-7760.md

```diff
@@ -0,0 +1,511 @@
+
+## Motivation
+
+Having standardized minimal bytecode for upgradeable proxies enables the following:
+
+1. Automatic verification on block explorers.
+2. Ability for immutable arguments to be queried onchain, as these arguments are stored at the same bytecode offset,
+3. Ability for the implementation to be queried and verified onchain.
+
+The minimal nature of the proxies enables cheaper deployment and runtime costs.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Overview
+
+All of the following proxies MAY have optional data bytecode appended to the end of their runtime bytecode. 
+
+Emitting the ERC-1967 events during initialization is OPTIONAL. Indexers MUST NOT expect the initialization code to emit the ERC-1967 events.
+
+### Minimal ERC-1967 transparent upgradeable proxy
+
+The transparent upgradeable proxy MUST be deployed by a factory that is responsible for authenticating upgrades.
+
+As the proxy's runtime bytecode contains logic to allow the factory to set any storage slot with any value, the initialization code MAY skip storing the implementation slot.
+
+The upgrading logic does not emit the ERC-1967 event. Indexers MUST NOT expect the upgrading logic to emit the ERC-1967 events.
+
+During upgrades, the factory MUST call the upgradeable proxy with following calldata:
+
+```solidity
+abi.encodePacked(
+    // The new implementation address, converted to a 32-byte word.
+    uint256(uint160(implementation)),
+    // ERC-1967 implementation slot.
+    bytes32(0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc),
+    // Optional calldata to be forwarded to the implementation
+    // via delegatecall after setting the implementation slot.
+    ""
+)
+```
+
+As it is beneficial to install the factory at a vanity address with leading zero bytes so that the proxy's bytecode can be optimized to be shorter, variants for 14-byte factory address are provided.
+
+#### Minimal ERC-1967 transparent upgradeable proxy (20-byte factory address regular variant)
+
+This is the runtime bytecode:
+
+```
+3d3d336d________________________________________14605157363d3d37363d7f360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc545af43d6000803e604c573d6000fd5b3d6000f35b3d3560203555604080361115604c5736038060403d373d3d355af43d6000803e604c573d6000fd
+```
+
+where `________________________________________` is the 20-byte factory address.
+
+
+#### Minimal ERC-1967 transparent upgradeable proxy (14-byte factory address regular variant)
```

Comment:
```suggestion
#### 14-byte factory address variant of minimal ERC-1967 transparent upgradeable proxy
```

Reply:
Here and below, I think the subsection names of specifications are important as one of the key proposal in this ERC is to propose different forms of minimal proxies. Current naming of these sections are a bit confusing I think, suggest revisit for better readability